### PR TITLE
Make Oak Functions cloud run service private

### PR DESCRIPTION
**Pull request by rbehjati**
* Changes Oak Functions Cloud Run service to require authentication 
* Copies the correct `.wasm` file to the example docker image
* Fixes the example

### CODE REVIEW

File: scripts/deploy_oak_functions_endpoints [hunk 1]
```diff
@@ -86,5 +86,5 @@ gcloud beta run deploy "${CLOUD_RUN_SERVICE_NAME}" \
   --image="gcr.io/oak-ci/endpoints-runtime-serverless:${IMAGE_TAG}" \
   --use-http2 \
   --allow-unauthenticated \
-  --platform managed \
-  --project=oak-ci
+  --platform=managed \
+  --project=oak-ci
```
Comment (conradgrobler) [inline scripts/deploy_oak_functions_endpoints:90]: Super nit: extra trailing space added

File: scripts/deploy_oak_functions_loader [hunk 2]
```diff
@@ -46,11 +50,12 @@ gcloud auth activate-service-account \
...
-  --allow-unauthenticated \
+  --no-allow-unauthenticated \
   --use-http2 \
   --concurrency=20 \
   --memory=8G \
   --cpu=4 \
   --min-instances=1 \
   --max-instances=10 \
-  --platform=managed
+  --platform=managed
```
Comment (conradgrobler) [inline scripts/deploy_oak_functions_loader:60]: Super nit: extra line break.

File: oak_functions/examples/weather_lookup/example.toml [hunk 3]
```diff
@@ -16,6 +16,5 @@ additional_args = [
 rust = { Cargo = { cargo_manifest = "oak_functions/client/rust/Cargo.toml" }, additional_args = [
   "--uri=http://localhost:8080",
   "--request={\"lat\":52,\"lon\":0}",
-  "--expected-response={\"temperature_degrees_celsius\":16}"
```
Comment (conradgrobler) [inline oak_functions/examples/weather_lookup/example.toml:19]: Not part of this PR, but we should figure out a better way for testing this. The java client still checks for the expected value, and it changes every time the lookup data is regenerated.

File: oak_functions/examples/weather_lookup/scripts/cloud_run_deploy [hunk 4]
```diff
@@ -12,13 +12,43 @@ export EXAMPLE_NAME='weather_lookup'
 export FUNCTIONS_INSTANCE_NAME='oak-functions-weather-lookup'
 export CLOUD_RUN_SERVICE_NAME='weather-lookup-endpoint'
 export API_TITLE='Oak Functions Weather Lookup Endpoint'
-
+export CONNECTOR_NAME="weather-lookup-connector"
+
```
Comment (conradgrobler) [inline oak_functions/examples/weather_lookup/scripts/cloud_run_deploy:16]: Super nit:  in the interest of consistency, a single blank line as a separator would be preferred.

**Review by conradgrobler — APPROVED** [review]
Thanks. I think it might be a good idea to create an issue to keep track of the need to also configure internal-only ingress and a VPC network for additional protection in future.

**Review by rbehjati — COMMENTED** [review]
Thanks for the review.

**Comment by rbehjati** [comment]
> Thanks. I think it might be a good idea to create an issue to keep track of the need to also configure internal-only ingress and a VPC network for additional protection in future.

Created #2209 to track this.
